### ISSUE TITLE
GCM Causing crash in signed apk 

### ISSUE THREAD

**Issue by FelipeFMMobile** [infobip/mobile-messaging-sdk-android#12]
We are facing this issue in or signed apk, below the stack trace. 
 

  gradleVersion = '3.4.0'
  kotlinVersion = "1.3.31"

```
2019-05-02 11:43:57.419 8679-8679/br.com.meupag E/AndroidRuntime: FATAL EXCEPTION: main
    Process: br.com.meupag, PID: 8679
    java.lang.AssertionError: java.lang.NoSuchFieldException: GCM
        at com.google.gson.internal.bind.TypeAdapters$a.<init>(SourceFile:12)
        at com.google.gson.internal.bind.TypeAdapters$30.a(SourceFile:5)
        at com.google.gson.Gson.a(SourceFile:37)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.a(SourceFile:35)
        at com.google.gson.Gson.a(SourceFile:37)
        at com.google.gson.Gson.a(SourceFile:10)
        at com.google.gson.Gson.a(SourceFile:7)
        at com.google.gson.Gson.a(SourceFile:55)
        at com.google.gson.Gson.a(SourceFile:53)
        at org.infobip.mobile.messaging.api.support.http.serialization.JsonSerializer.serialize(SourceFile:1)
        at org.infobip.mobile.messaging.InstallationMapper.a(SourceFile:46)
        at org.infobip.mobile.messaging.platform.AndroidBroadcaster.a(SourceFile:25)
        at org.infobip.mobile.messaging.mobile.appinstance.InstallationSynchronizer$2.after(SourceFile:11)
        at org.infobip.mobile.messaging.mobile.appinstance.InstallationSynchronizer$2.after(SourceFile:1)
        at org.infobip.mobile.messaging.mobile.common.MRetryableTask$1.after(SourceFile:1)
        at org.infobip.mobile.messaging.mobile.common.MAsyncTask$2.onPostExecute(SourceFile:6)
        at org.infobip.mobile.messaging.mobile.common.MAsyncTask$2.onPostExecute(SourceFile:1)
        at android.os.AsyncTask.finish(AsyncTask.java:695)
        at android.os.AsyncTask.access$600(AsyncTask.java:180)
        at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:712)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6669)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.lang.NoSuchFieldException: GCM
        at java.lang.Class.getField(Class.java:1601)
        at com.google.gson.internal.bind.TypeAdapters$a.<init>(SourceFile:6)
        at com.google.gson.internal.bind.TypeAdapters$30.a(SourceFile:5) 
        at com.google.gson.Gson.a(SourceFile:37) 
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.a(SourceFile:35) 
        at com.google.gson.Gson.a(SourceFile:37) 
        at com.google.gson.Gson.a(SourceFile:10) 
        at com.google.gson.Gson.a(SourceFile:7) 
        at com.google.gson.Gson.a(SourceFile:55) 
        at com.google.gson.Gson.a(SourceFile:53) 
        at org.infobip.mobile.messaging.api.support.http.serialization.JsonSerializer.serialize(SourceFile:1) 
        at org.infobip.mobile.messaging.InstallationMapper.a(SourceFile:46) 
        at org.infobip.mobile.messaging.platform.AndroidBroadcaster.a(SourceFile:25) 
        at org.infobip.mobile.messaging.mobile.appinstance.InstallationSynchronizer$2.after(SourceFile:11) 
        at org.infobip.mobile.messaging.mobile.appinstance.InstallationSynchronizer$2.after(SourceFile:1) 
        at org.infobip.mobile.messaging.mobile.common.MRetryableTask$1.after(SourceFile:1) 
        at org.infobip.mobile.messaging.mobile.common.MAsyncTask$2.onPostExecute(SourceFile:6) 
        at org.infobip.mobile.messaging.mobile.common.MAsyncTask$2.onPostExecute(SourceFile:1) 
        at android.os.AsyncTask.finish(AsyncTask.java:695) 
        at android.os.AsyncTask.access$600(AsyncTask.java:180) 
        at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:712) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:6669) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858) 
2019-05-02 11:43:57.756 8679-8785/br.com.meupag I/CrashlyticsCore: Crashlytics report upload complete: 5CCB01830353-0001-20FC-F0392E44BF98
2019-05-02 11:43:57.849 8679-8743/br.com.meupag E/CrashlyticsCore: Error occurred sending report prog.lo@ece0bf
    io.fabric.sdk.android.services.network.HttpRequest$HttpRequestException: java.io.FileNotFoundException: /data/user/0/br.com.meupag/files/.Fabric/com.crashlytics.sdk.android.crashlytics-core/fatal-sessions/5CCB01830353-0001-20FC-F0392E44BF98.cls (No such file or directory)
        at io.fabric.sdk.android.services.network.HttpRequest.a(SourceFile:67)
        at prog.Tm.a(SourceFile:64)
        at prog.io.a(SourceFile:12)
        at prog.Cn$k.run(SourceFile:3)
        at prog.Ym.run(SourceFile:1)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:458)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at prog.QQ.a(SourceFile:1)
        at io.fabric.sdk.android.services.common.BackgroundPriorityRunnable.run(SourceFile:2)
        at java.lang.Thread.run(Thread.java:764)
     Caused by: java.io.FileNotFoundException: /data/user/0/br.com.meupag/files/.Fabric/com.crashlytics.sdk.android.crashlytics-core/fatal-sessions/5CCB01830353-0001-20FC-F0392E44BF98.cls (No such file or directory)
        at java.io.FileInputStream.open0(Native Method)
        at java.io.FileInputStream.open(FileInputStream.java:231)
        at java.io.FileInputStream.<init>(FileInputStream.java:165)
        at io.fabric.sdk.android.services.network.HttpRequest.a(SourceFile:64)
        at prog.Tm.a(SourceFile:64) 
        at prog.io.a(SourceFile:12) 
        at prog.Cn$k.run(SourceFile:3) 
        at prog.Ym.run(SourceFile:1) 
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:458) 
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at prog.QQ.a(SourceFile:1) 
        at io.fabric.sdk.android.services.common.BackgroundPriorityRunnable.run(SourceFile:2) 
        at java.lang.Thread.run(Thread.java:764) 
```

**Comment by tjuric** [comment]
Hi @FelipeFMMobile,

Our support informed us that this issue happens with combination of `com.android.tools.build:gradle` version 3.4.0 and the gradle wrapper 5.1.1, but downgrading it to 3.3.2 and wrapper to 4.10.1 the issue is not present. Can you confirm this behaviour?

We'll make our compatibility checks and inform you on our findings.

Regards, Tereza

**Comment by FelipeFMMobile** [comment]
Yes Tereza,

We discover that downgrading solved.

Please let me know when you fix it in the next releases.

Best

Felipe

Obter o Outlook para iOS<https://aka.ms/o0ukef>
________________________________
De: Tereza Juric <notifications@github.com>
Enviado: Sunday, May 5, 2019 7:12:50 AM
Para: infobip/mobile-messaging-sdk-android
Cc: FelipeFMMobile; Mention
Assunto: Re: [infobip/mobile-messaging-sdk-android] GCM Causing crash in signed apk (#12)


Hi @FelipeFMMobile<https://github.com/FelipeFMMobile>,

Our support informed us that this issue happens with combination of com.android.tools.build:gradle version 3.4.0 and the gradle wrapper from 5.1.1, but downgrading it to 3.3.2 and wrapper to 4.10.1 the issue is not present. Can you confirm this behaviour?

We'll make our compatibility checks and inform you on our findings.

Regards, Tereza

**Comment by tjuric** [comment]
Release is published within [2.0.14](https://github.com/infobip/mobile-messaging-sdk-android/releases/tag/2.0.14) SDK. 
Please, retest it and notify us if you find any other pending issues on your side.

Regards, 
Tereza

**Comment by tjuric** [comment]
We've been informed by our support that the issue was solved. Closing it here also. Please, reopen it if it's needed.

Regards,
Tereza
